src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPoolBlockingQueue.AnyOS.cs

```diff
@@ -0,0 +1,219 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+using Internal.Runtime.CompilerServices;
+
+namespace System.Threading
+{
+    public static partial class ThreadPool
+    {
+        internal static void RecordBlockingCallsite()
+            => ThreadPoolBlockingQueue.Enable();
+    }
+
+    internal class ThreadPoolBlockingQueue : IThreadPoolWorkItem
+    {
+        [ThreadStatic]
+        private static object? t_currentWorkItem;
+
+        private static ThreadPoolBlockingQueue? s_processor;
+        private static HashSet<object>? s_blockingItems;
+        private static bool s_enabled;
+
+        public static bool IsEnabled => s_enabled;
+
+        private readonly ConcurrentQueue<object> _workItems = new ConcurrentQueue<object>();
```

Comment:
```suggestion
        private readonly ConcurrentQueue<object> _workItems = new();
```

Are you able to use C# 9, implicit target type new here?